### DISCUSSION TITLE
Can't find stylesheet to import

### DISCUSSION THREAD

**Discussion by Tom-Whi** [cotes2020/jekyll-theme-chirpy#1809] [Q&A]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Description

I have merged with the upstream repo and I am getting the following error during the build process of Deploy Jekyll site to Pages action. 

https://github.com/Tom-Whi/tom-whi.github.io/actions/runs/9530555828/job/26270566378

```
Error: Can't find stylesheet to import.
  ╷
1 │ @import 'dist/bootstrap';
  │         ^^^^^^^^^^^^^^^^
  ╵
  main.bundle.scss 1:9                                                                           @import
  /home/runner/work/tom-whi.github.io/tom-whi.github.io/assets/css/jekyll-theme-chirpy.scss 1:9  root stylesheet 
  Conversion error: Jekyll::Converters::Scss encountered an error while converting 'assets/css/jekyll-theme-chirpy.scss':
                    Can't find stylesheet to import.
                    ------------------------------------------------
      Jekyll 4.3.3   Please append `--trace` to the `build` command 
                     for any additional information or backtrace. 
                    ------------------------------------------------
```

This happens because the jekyll-theme-chirpy.scss was updated to include an if/else statement for the environment if it equals "production" to use the _sass/main.bundle.scss.    Inside that file is referenced dist/bootstrap which I don't have (I guess?!)

I have built this from the starter, and I only just VSCode and github to develop it, I don't run the site locally for testing. 

### Operations you have already tried

I have tried removing the if/else statement from the file and the action completes without errors, but no longer displays new posts when I post them.

**Comment by heomne** [comment]
Solved this issue? i have a same thing

**Reply by ghost** [reply]
I was about to answer that
In the config.yml file writing correct url fixed these for me.
But tom whi didn't create using chirpy-starter like he mentioned above, and you also forked the main repo to create the site instead of chirpy starter, so can't tell about these cases.

**Reply by roooocks** [reply]
**npm run build**
I didn't make it into starter, but there was the same error. I solved it the way above

**Reply by Tom-Whi** [reply]
@B-and-B-Studio I did use the Chripy Starter (at least I think I did, it was a while ago).  

My comments above start by saying that to keep it up to date I am trying to merge from the upstream repo.  I have got the correct URL in the config file but I am still getting the error in the original post. This is my config showing the correct URL based on the format in the comment. 

![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/19737487/d58f538f-9a37-4041-880f-5fff119ee127)

**Reply by ghost** [reply]
@Tom-Whi hey I checked your repo again.

![Screenshot_20240711-153439](https://github.com/cotes2020/jekyll-theme-chirpy/assets/174127938/67f585f3-3616-4720-ada5-f2ef8a31c80d)

You are using the fork of main theme repo `jekyll-theme-chirpy`, and not the `chirpy-starter` template. And as someone who doesn't have much knowledge of Jekyll/ruby/yml I can only say it's a hassle to use it this way. Whereas using starter template you just need to change a few lines at most to update it.

So I just override layouts/includes when needed within the starter template.

You can check [this repo of mine](https://github.com/B-and-B-Studio/Roadmap) for reference, just know that I've moved on to local developing yesterday because of the live preview feature(it helped me progress faster) and my website is 95% complete now within a few hours from the design/customisation PoV.
**Nonetheless this repo should still help you with how to do things because everything in it were completely developed just using GitHub from my browser**. 
I'll upload the final version of my site at my [SoumyaK4](https://soumyak4.github.io) site within a few days.

Also here I checked your repo and the last tag it had was in version 5 whereas the latest version is 7.

_So I'm assuming something went wrong while you were merging with the upstream repo_

![Screenshot_20240711-152917](https://github.com/cotes2020/jekyll-theme-chirpy/assets/174127938/73b8e6d9-7dec-4d6a-9338-06485ceb955c)

> I'd recommend you download all your files in `_posts`, and any other `_tabs` files you added/modified, then just delete the current repo and start with the `chirpy-starter` and upload those downloaded files again at specific directories.

Also, can you try removing [line 170-172 in your config](https://github.com/Tom-Whi/tom-whi.github.io/blob/64162465879e407aa2a5b6ed7398257f2a804973/_config.yml#L170)? Cause I don't see these lines in the latest version, maybe issues because of this. 

Honestly that's all the help I can give as my knowledge stands now.

Hope this turns out helpful to you.
Have a great day.
Good Luck

**Reply by Tom-Whi** [reply]
Hey @B-and-B-Studio , thank you so much for the help!  Looked back at my notes and it's not clear how I started, however it's clear that mine says "forked from" and yours says "generated from" ...  So I think I didn't do it in the way I thought I did it :-) 

I think you're right that I'm also not getting the latest tags either and mine are not up to date so that might be why the update isn't working as I expected.  I think my limited undersanding of Git has failed me and that's why I'm not forking the repo. 

I am also someone who doesn't do much with Jekyll/ruby/yml either, so I wanted a nice and easy setup.  I think I might look to trash my site and start it again based on what you've shown me...  It makes way more sense and I think I pinned myself in a corner by forking the repo.  Thank you for suggesting removing those lines - I won't be able to try this as I've broken my actions beyond repair so I think starting again is the best for me. 

Thank you again for taking the time to compare the repo's and share your knowledge.

**Reply by ghost** [reply]
@Tom-Whi great!!! Just download your post files, then start with chirpy starter and upload your fils inside posts.
> You can actually download the whole repo as zip, that way there would be a backup of all files.

**Comment by wwwGeneral** [comment]
Hi ! I have the same error as @Tom-Whi , however as i'm not using chirpy start but the original one (i just merged new updates) i don't know what to do.
What is strange is that i got no error on my computer when i build but when github workflow do it, it crashes :

```
Run bundle exec jekyll b -d "_site --trace"
  bundle exec jekyll b -d "_site --trace"
  shell: /usr/bin/bash -e {0}
  env:
    GITHUB_PAGES: true
    JEKYLL_ENV: production
Configuration file: /home/runner/work/wwwGeneral.github.io/wwwGeneral.github.io/_config.yml
 Theme Config file: /home/runner/work/wwwGeneral.github.io/wwwGeneral.github.io/_config.yml
            Source: /home/runner/work/wwwGeneral.github.io/wwwGeneral.github.io
       Destination: /home/runner/work/wwwGeneral.github.io/wwwGeneral.github.io/_site --trace
 Incremental build: disabled. Enable with --incremental
      Generating... 
Error: Can't find stylesheet to import.
  ╷
1 │ @import 'dist/bootstrap';
  │         ^^^^^^^^^^^^^^^^
  ╵
  main.bundle.scss 1:9                                                                                 @import
  /home/runner/work/wwwGeneral.github.io/wwwGeneral.github.io/assets/css/jekyll-theme-chirpy.scss 1:9  root stylesheet 
  Conversion error: Jekyll::Converters::Scss encountered an error while converting 'assets/css/jekyll-theme-chirpy.scss':
                    Can't find stylesheet to import.
                    ------------------------------------------------
      Jekyll 4.3.3   Please append `--trace` to the `build` command 
                     for any additional information or backtrace. 
                    ------------------------------------------------
/home/runner/work/wwwGeneral.github.io/wwwGeneral.github.io/vendor/bundle/ruby/3.[2](https://github.com/wwwGeneral/wwwGeneral.github.io/actions/runs/10529458702/job/29177272291#step:5:2).0/gems/jekyll-sass-converter-3.0.0/lib/jekyll/converters/scss.rb:175:in `rescue in convert': Can't find stylesheet to import. (Jekyll::Converters::Scss::SyntaxError)

        raise SyntaxError, e.message
              ^^^^^^^^^^^^^^^^^^^^^^
	from /home/runner/work/wwwGeneral.github.io/wwwGeneral.github.io/vendor/bundle/ruby/[3](https://github.com/wwwGeneral/wwwGeneral.github.io/actions/runs/10529458702/job/29177272291#step:5:3).2.0/gems/jekyll-sass-converter-3.0.0/lib/jekyll/converters/scss.rb:1[5](https://github.com/wwwGeneral/wwwGeneral.github.io/actions/runs/10529458702/job/29177272291#step:5:5)9:in `convert'
```

Did anyone succeeded in fixing it ?

**Reply by Naminar** [reply]
I have the same problem. When i'm trying to do localy - everything is okay!

![image](https://github.com/user-attachments/assets/2500a269-621b-4296-9456-4f8296a8c8a2)

**Comment by wwwGeneral** [comment]
I just figured it out, there is a new update procedure :  
https://github.com/cotes2020/jekyll-theme-chirpy/wiki/Upgrade-Guide

**Comment by Gaur4vGaur** [comment]
Yes, you need to run 
`npm run build`

followed by
`git add assets/js/dist _sass/dist -f`

**Comment by lelalela** [comment]
.github\workflows\jekyll.yml에 build에서 Build with Jekyll 위에
- name: npm build
        run: npm install && npm run build
넣고 해보세요 전 이렇게 해결했습니다.

**Reply by rse112** [reply]
Thank you so much for your suggestion! I've been struggling with this issue for almost 4 hours, and your advice finally solved it. I really appreciate your help!

**Reply by minju0611** [reply]
you are my sunshine

Thanks a lot.

**Reply by JunghyunHwang** [reply]
이 글은 모두가 알아야 합니다! 감사합니다!

**Reply by sebinyday** [reply]
If is no such file as .github\workflows\jekyl.yml.
add code 
.github\workflows\pages-deploy.yml
![image](https://github.com/user-attachments/assets/342c0c01-726d-494a-9ed2-c489715e8bc6)
like this. 

find `build site` and add 
```
- name: npm build
        run: npm install && npm run build
```

**Reply by HyunJinNo** [reply]
감사합니다. 덕분에 문제를 해결할 수 있었습니다.

**Reply by ghostdavid** [reply]
@sebinyday @lelalela I'm so grateful, as a beginner, I was on the verge of despair trying to solve this issue. Thank you so much for seeing and responding to your replies. I'm so grateful, I love you all.

**Reply by Synvani** [reply]
@sebinyday @lelalela thanks a lot! I documented everything needed to successfully deploy a forked chirpy in my blog, hopefully it will help others encountering this in the near future https://synvan.com/posts/kickstarting-chirpy/

**Comment by jinhg0214** [comment]
```
Error: Can't find stylesheet to import.
  ╷
1 │ @use 'vendors/bootstrap';
  │ ^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
  main.bundle.scss 1:1                                                                               @use
  /home/runner/work/jinhg02[14](https://github.com/jinhg0214/jinhg0214.github.io/actions/runs/13384505203/job/37378700699#step:7:15).github.io/jinhg0214.github.io/assets/css/jekyll-theme-chirpy.scss 1:1  root stylesheet 
  Conversion error: Jekyll::Converters::Scss encountered an error while converting 'assets/css/jekyll-theme-chirpy.scss':
                    Can't find stylesheet to import.
                    ------------------------------------------------
      Jekyll 4.4.1   Please append `--trace` to the `build` command 
                     for any additional information or backtrace. 
                    ------------------------------------------------
```
I got the following error, and I solved it by the way below. Hope it helps.

`git add assets/js/dist/ _sass/vendors/ -f`

**Reply by vinayak19th** [reply]
Thanks for the solution ! 
Just a note, a better longer term solution would be to remove `_sass/vendors/` from the .gitignore rather than force add it
